oshdb-filter/src/main/java/org/heigit/ohsome/oshdb/filter/ChangesetIdFilterEquals.java

```diff
@@ -0,0 +1,43 @@
+package org.heigit.ohsome.oshdb.filter;
+
+import org.heigit.ohsome.oshdb.osm.OSMEntity;
+import org.heigit.ohsome.oshdb.util.mappable.OSMContribution;
+import org.heigit.ohsome.oshdb.util.mappable.OSMEntitySnapshot;
+import org.jetbrains.annotations.Contract;
+
+/**
+ * A filter which selects OSM contributions by a changeset id.
+ */
+public class ChangesetIdFilterEquals extends NegatableFilter {
+  final long changesetId;
+
+  ChangesetIdFilterEquals(long changesetId) {
+    super(new FilterInternal() {
+      @Override
+      public boolean applyOSM(OSMEntity entity) {
+        return true;
+      }
```

Comment:
good point. I left this out initially, because I thought this would just essentially duplicate what the cell iterator does already internally while creating the contribution objects, but I forgot that this early check in the OSH stage can skip quite a lot of expensive geometry calculations if a query does filter only on changeset ids, for example. :+1: 

I will take your proposed approach, with just a few tweaks (e.g. there was a small error in line 22: it should be `way.getNodes().…` instead of `entitiy.…` :wink: )

Reply:
How should we proced with the throwing exception from (getNode/getWay) I will remove them, and can already do it in this PR if you like. Let me know, if i should commit into this PR.